### PR TITLE
fix(oracle): fix error handling on sync query

### PR DESCRIPTION
**Pull request by paulozulato**
Fixes https://emqx.atlassian.net/browse/EMQX-10075

### <samp>🤖 Generated by Copilot at 8a14bac</samp>

Improve error handling and consistency in `emqx_oracle` module. Return different error types for recoverable and unrecoverable errors, and use a common error format for both.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [ ] ~Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files~
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- [ ] ~Schema changes are backward compatible~

## Checklist for CI (.github/workflows) changes

- [ ] ~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- [ ] ~Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

File: apps/emqx_oracle/src/emqx_oracle.erl [hunk 1]
```diff
@@ -146,7 +146,14 @@ on_query(
     Type = query,
     {NameOrSQL2, Data} = proc_sql_params(TypeOrKey, NameOrSQL, Params, State),
     Res = on_sql_query(InstId, PoolName, Type, ?SYNC_QUERY_MODE, NameOrSQL2, Data),
-    handle_result(Res).
+    case Res of
+        {error, {recoverable_error, _Error}} ->
+            Res;
+        {error, {unrecoverable_error, {_RetCode, _Reason}}} ->
+            Res;
```
Comment (thalesmg) [inline apps/emqx_oracle/src/emqx_oracle.erl:153]: Maybe we could add those clauses to `handle_result`?

File: apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl [hunk 2]
```diff
@@ -167,17 +167,25 @@ sql_create_table() ->
...
     ResourceId = resource_id(Config),
-    _ = emqx_resource:simple_sync_query(ResourceId, {sql, sql_drop_table()}),
+    drop_table_if_exist(Config),
     {ok, [{proc_result, 0, _}]} = emqx_resource:simple_sync_query(
         ResourceId, {sql, sql_create_table()}
     ),
     ok.
 
-drop_table(Config) ->
+drop_table_if_exist(Config) ->
```
Comment (thalesmg) [inline apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl:181]:
```suggestion
drop_table_if_exists(Config) ->
```

File: apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl [hunk 3]
```diff
@@ -167,17 +167,25 @@ sql_create_table() ->
...
     ResourceId = resource_id(Config),
-    emqx_resource:simple_sync_query(ResourceId, {query, sql_drop_table()}),
+    {ok, [{result_set, [<<"COUNT(*)">>], [], [[{TableExist}]]}]} =
+        emqx_resource:simple_sync_query(ResourceId, {query, sql_check_table_exist()}),
+    case TableExist of
+        0 -> ok;
+        1 -> emqx_resource:simple_sync_query(ResourceId, {query, sql_drop_table()})
```
Comment (thalesmg) [inline apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl:187]: Q: doesn't Oracle have `drop table if exists xxx` like i.e. [Postgres](https://www.postgresql.org/docs/current/sql-droptable.html)?

File: apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl (hunk 2 above)
Comment (paulozulato) [reply apps/emqx_bridge_oracle/test/emqx_bridge_oracle_SUITE.erl:181]: Ack

File: apps/emqx_oracle/src/emqx_oracle.erl (hunk 1 above)
Comment (paulozulato) [reply apps/emqx_oracle/src/emqx_oracle.erl:153]: it could be...

File: apps/emqx_oracle/src/emqx_oracle.erl [hunk 4]
```diff
@@ -346,6 +346,10 @@ to_bin(Bin) when is_binary(Bin) ->
 to_bin(Atom) when is_atom(Atom) ->
     erlang:atom_to_binary(Atom).
 
+handle_result({error, {recoverable_error, _Error}} = Res) ->
+    Res;
+handle_result({error, {unrecoverable_error, {_RetCode, _Reason}}} = Res) ->
```
Comment (thalesmg) [inline apps/emqx_oracle/src/emqx_oracle.erl:351]:
```suggestion
handle_result({error, {unrecoverable_error, _}} = Res) ->
```

**Review by zmstone — APPROVED** [review]

**Review by thalesmg — APPROVED** [review]

**Pull request merged**